core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -5666,22 +5663,30 @@ class ReplicaManagerTest {
   }
 
   private def topicsCreateDelta(startId: Int, isStartIdLeader: Boolean, partition:Int = 0, directoryIds: List[Uuid] = List.empty): TopicsDelta = {
+    topicsCreateDelta(startId, isStartIdLeader, List(partition), directoryIds, "foo", FOO_UUID)
+  }
+
+  private def topicsCreateDelta(startId: Int, isStartIdLeader: Boolean, partitions: List[Int],
+                                directoryIds: List[Uuid], topicName: String, topicId: Uuid): TopicsDelta = {
```

Comment:
What do you think about this adjustment
```suggestion
  private def topicsCreateDelta(startId: Int, isStartIdLeader: Boolean, partitions: List[Int] = List(0),
                                directoryIds: List[Uuid] = List.empty, topicName: String = "foo", topicId: Uuid = FOO_UUID): TopicsDelta = {
```

Reply:
I used overloading here since a few existing calls would break due to mismatched params if we used default args. But if you think default params make it cleaner overall, happy to switch—what do you think?

Reply:
> I used overloading here since a few existing calls would break due to mismatched params if we used default args

I think updating the existing calls is totally fine.

Reply:
Thanks for the feedback! Just updated it.